### DISCUSSION TITLE
自动获取ip地址

### DISCUSSION THREAD

**Discussion by koszzz** [messense/aliyundrive-webdav#261] [Q&A] [answered]
我是在软路由上使用的，如果使用默认填写的0.0.0.0地址或者使用ipv6的::1都是不行的，必须填写软路由自身的ip地址。
如果是ipv4地址好说，把ipv4内网地址填写进去就行。但使用ipv6时，填写软路由自身的ipv6地址虽然可行，但ipv6地址是会更新变动的，就挺不方便的。
建议作者增加自动获取ip地址的功能

**Comment by messense** [comment, marked as answer]
> 如果使用默认填写的0.0.0.0地址或者使用ipv6的::1都是不行的，必须填写软路由自身的ip地址

为什么呢，有什么报错信息吗？

**Reply by koszzz** [reply]
没有报错，刚才尝试用docker版的日志写的是监听0.0.0.0:8080，这样的话ipv4访问没有问题，ipv6不行

**Reply by koszzz** [reply]
dd，docker版的话要修改监听地址是通过环境变量修改吗？是的话是哪个环境变量？

**Reply by messense** [reply]
~~`PORT`~~ `HOST`，ipv6: https://github.com/messense/aliyundrive-webdav/discussions/159

**Reply by koszzz** [reply]
![image](https://user-images.githubusercontent.com/78025085/150107855-07714824-1a82-42c6-98ca-2be4d93280bb.png)
PORT环境变量是修改端口号的，我是想修改监听地址wwww

**Reply by messense** [reply]
`HOST`

文档：https://github.com/messense/aliyundrive-webdav#%E5%91%BD%E4%BB%A4%E8%A1%8C%E7%94%A8%E6%B3%95

**Reply by koszzz** [reply]
HOST设置成了::日志里还是监听的0.0.0.0

**Reply by koszzz** [reply]
HOST设置成1.1.1.1日志里监听的都还是0.0.0.0

**Reply by messense** [reply]
确认下环境变量生效了没有。

**Reply by messense** [reply]
BTW，问题反馈请给出复现步骤。

**Reply by koszzz** [reply]
同时在docker里添加两个环境变量：HOST=1.1.1.1,PORT=2333
日志里监听的是0.0.0.0:2333
![image](https://user-images.githubusercontent.com/78025085/150112828-95056362-574f-4eaf-866f-62b298b3e82d.png)

![image](https://user-images.githubusercontent.com/78025085/150111703-afc1b3a6-28b1-4013-8370-7964cdca846a.png)

**Reply by messense** [reply]
等 https://github.com/messense/aliyundrive-webdav/actions/runs/1717483706 这个跑完，用 `messense/aliyundrive-webdav:main` 镜像再试一下。

**Reply by koszzz** [reply]
同样的问题，没有变化

**Reply by messense** [reply]
无法复现，建议确认是用的 messense/aliyundrive-webdav:main 这个镜像，注意是 main tag

```bash
❯ docker run --name=aliyundrive-webdav \
  -e REFRESH_TOKEN=xxx \
  -e HOST=:: \
  messense/aliyundrive-webdav:main

2022-01-19T12:10:29.256789Z  INFO aliyundrive_webdav::drive: refresh token succeed refresh_token=xxx nick_name=messense
2022-01-19T12:10:29.257935Z  INFO aliyundrive_webdav::drive: found default drive drive_id=49804381
2022-01-19T12:10:29.258902Z  INFO aliyundrive_webdav: listening on http://[::]:8080
```

**Reply by koszzz** [reply]
感谢大佬，更换镜像后没有问题了

**Comment by xiaoLit** [comment]
我也是类似问题  我这里的IPv4地址是公网IP
每次重启路由器后都会变化  然后infuse添加源是不能修改ip地址的  
又需要重新删除添加源  重新下载数据等非常麻烦

**Reply by xiaoLit** [reply]
> 我也是类似问题 我这里的IPv4地址是公网IP
> 每次重启路由器后都会变化 然后infuse添加源是不能修改ip地址的
> 又需要重新删除添加源 重新下载数据等非常麻烦

抱歉是菜鸟  用了ddns就可以了
